src/app/services/socket.service.js

```diff
@@ -69,6 +71,7 @@ class SocketService {
 
   emit(eventName, data, callback) {
     //this.$log.debug('emit', eventName);
+    this.loadingBar.start();
```

Comment:
This code assumes that every event emitted to the backend will have a corresponding event in response to the UI. Is there any way to enforce or validate this assumption?

Reply:
Not every event emitted will have a corresponding response to the UI. And actually most emits from the BE to UI are not invoked by the UI itself.
IF your goal was to signal we were loading the library, you will better look into library browsing js

Reply:
In my testing, this code worked without issue. Events emitted from the backed to the UI without being invoked by the UI should pose no problem. In the case where there are events emitted by the UI where there is no corresponding response, it is possible to add exceptions for those scenarios.

Reply:
An possible alternative would be to create a whitelist of events which we want to show the loading bar for.

Reply:
Whitelist is a good idea IMHO!

Reply:
There is no complete doc unfortunately, but you can see all the events in the websocket plugin.
See:
https://github.com/volumio/Volumio2/blob/master/app/plugins/user_interface/websocket/index.js

IMHO the two events where which we can start are:
browseLibrary
and 
search

What do you think?